test/unit/ownership_test.rb

```diff
@@ -46,28 +47,168 @@ class OwnershipTest < ActiveSupport::TestCase
       @ownerships = @user.ownerships.by_indexed_gem_name
     end
 
-    should "ownwerships should be sorted by rubygem name ascedent order" do
+    should "ownerships should be sorted by rubygem name ascendant order" do
       assert_equal @gems.sort, (@ownerships.map { |own| own.rubygem.name })
     end
   end
 
-  context "#safe_destroy" do
+  context "#destroy_and_notify" do
     setup do
       @rubygem       = create(:rubygem)
       @ownership_one = create(:ownership, rubygem: @rubygem)
       @ownership_two = create(:ownership, rubygem: @rubygem)
+      @ownership_three = create(:ownership, :unconfirmed, rubygem: @rubygem)
     end
 
     should "allow deletion of one ownership" do
-      @ownership_one.safe_destroy
+      @ownership_one.destroy_and_notify
       assert_equal 1, @rubygem.owners.length
+      assert_equal 2, @rubygem.owners_including_unconfirmed.length
+    end
+
+    should "allow deletion of unconfirmed ownership" do
+      @ownership_three.destroy_and_notify
+      assert_equal 2, @rubygem.owners_including_unconfirmed.length
     end
 
     should "not allow deletion of both ownerships" do
-      @ownership_one.safe_destroy
-      @ownership_two.safe_destroy
+      @ownership_one.destroy_and_notify
+      @ownership_two.destroy_and_notify
       assert_equal 1, @rubygem.owners.length
       assert_equal @ownership_two.user, @rubygem.owners.last
     end
   end
+
+  context "#create" do
+    setup do
+      @authorizer = create(:user)
+      @new_owner = create(:user)
+      @rubygem = create(:rubygem)
+    end
+
+    should "create unconfirmed ownership" do
+      ownership = @rubygem.ownerships.create(user: @new_owner, authorizer: @authorizer)
+      assert_nil ownership.confirmed_at
+    end
+
+    should "generate 20 char hex confirmation token" do
+      ownership = @rubygem.ownerships.create(user: @new_owner, authorizer: @authorizer)
+      assert_match(/[0-9a-f]{20}/, ownership.token)
+    end
+
+    should "not create without a user" do
+      ownership = @rubygem.ownerships.create(authorizer: @authorizer)
+      assert_equal ["must exist"], ownership.errors[:user]
+    end
+
+    should "not create without a rubygem" do
+      ownership = Ownership.create(user: @new_owner)
+      assert_equal ["must exist"], ownership.errors[:rubygem]
+    end
+
+    should "allow creation without authorizer" do
+      ownership = Ownership.create(user: @new_owner, rubygem: @rubygem)
+      assert_empty ownership.errors.full_messages
+    end
```

Comment:
Can we use `build` for the above test? It is nicer to use the least possible pre-condition to set up what we are testing. Also, add assert for `valid?`. check user_test.rb.